### ISSUE TITLE
MD051 should support arbitrary `id` attributes, not just `<a>` tags

### ISSUE THREAD

**Issue by MarkLodato** [DavidAnson/markdownlint#538]
MD051 currently only checks `<a>` tags for `id="..."` attributes, but this requirement seems arbitrary. Any HTML element may have an `id` attribute to define an anchor.

It seems like this can be easily fixed by removing the check for `a` tags on [line 58](https://github.com/DavidAnson/markdownlint/blob/cba2ca0dbdc8f11e4b97c054380ffd52bb7aae19/lib/md051.js#L58).

In my case, I have a table and use `id` attributes to link to specfic rows of the table:

```markdown
# Example

<table>
<tr id="foo"><td>Foo<td>Foo is good
<tr id="bar"><td>Bar<td>Bar is also good
</table>

I like [foo](#foo) and [bar](#bar).
```

```
$ npx markdownlint test.md
test.md:8:8 MD051/link-fragments Link fragments should be valid [Context: "[foo](#foo) and [bar](#bar)"]
test.md:8:8 MD051/link-fragments Link fragments should be valid [Context: "[foo](#foo)"]
```

Would you be willing to accept a PR that fixes this, or is there some other reason why only `a` tags are accepted?

**Comment by DavidAnson** [comment]
Yes, I would accept a PR for this! Please read the CONTRIBUTING guide first. Thank you!
